provider/pkg/await/await.go

```diff
@@ -90,17 +90,19 @@ type ReadConfig struct {
 
 type UpdateConfig struct {
 	ProviderConfig
-	Previous *unstructured.Unstructured
-	Inputs   *unstructured.Unstructured
-	Timeout  float64
-	Preview  bool
+	OldInputs  *unstructured.Unstructured
+	OldOutputs *unstructured.Unstructured
+	Inputs     *unstructured.Unstructured
+	Timeout    float64
+	Preview    bool
 	// IgnoreChanges is a list of fields to ignore when diffing the old and new objects.
 	IgnoreChanges []string
 }
 
 type DeleteConfig struct {
 	ProviderConfig
 	Inputs  *unstructured.Unstructured
+	Outputs *unstructured.Unstructured
```

Comment:
Important to note here that the `Inputs` field contained the _old outputs_ (`current`). Now it contains the old inputs. The deletion logic should therefore use `Outputs` to maintain the status-quo.  Note that `Inputs` is not actually used; kept only for symmetry.

A subtle (positive?) consequence is that the `timeout` annotation is drawn from the live object.